python/common/org/python/types/ByteArray.java

```diff
@@ -761,8 +761,16 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
     @org.python.Method(
             __doc__ = "B.isalnum() -> bool\n\nReturn True if all characters in B are alphanumeric\nand there is at least one character in B, False otherwise."
     )
-    public org.python.Object isalnum(java.util.List<org.python.Object> args, java.util.Map<java.lang.String, org.python.Object> kwargs, java.util.List<org.python.Object> default_args, java.util.Map<java.lang.String, org.python.Object> default_kwargs) {
-        throw new org.python.exceptions.NotImplementedError("bytearray.isalnum has not been implemented.");
+    public org.python.Object isalnum() {
+        if (this.value.length == 0) {
+            return new org.python.types.Bool(false);
+        }
+        for (byte ch: this.value) {
+            if (!Bytes._isalpha(ch) && !Bytes._isnum(ch)) {
```

Comment:
Is this correct `!Bytes._isalpha(ch) && !Bytes._isnum(ch)` ?

Should it not be more like:
`!(Bytes._isalpha(ch) && Bytes._isnum(ch))`

Reply:
It is correct because:
!AlphaNum = !(Alpha || Num) = !Alpha && !Num

But !(Alpha && Num) works too because any alphanum ch cannot be aplha and num at the same time. Your logic is slightly more efficient. Should I switch to your logic?

Reply:
On second thought @sziraqui its fine I think as you have it

Reply:
Cool! Thanks for reviewing.

Reply:
Ready to be merged?

Reply:
@eliasdorneles is the boss :-p

Reply:
ohk

Reply:
Also, you need to add tests!

Reply:
Already being tacked on https://github.com/pybee/voc/pull/753

Reply:
oops saw your message late. I had mentioned that I am working on these functions in issue 36. Hope someone will resolve the conflict

Reply:
So, this is already implemented in Bytes, see: https://github.com/pybee/voc/blob/78d2360c7f9f94daa2eae2c40a66d74f66926797/python/common/org/python/types/Bytes.java#L1053

For ByteArray, you just need to add tests and reuse that implementation.

Reply:
Ok. I thought these are supposed to be independent because they are non-static.

Reply:
@sziraqui I think I should have mentioned the pull request properly sorry, I have implemented in a similar way including the helper utility functions... and have done that for isdigit,isalpha,isalnum and used that in the ByteArray... even the test cases are done... waiting for the boss...

Reply:
Ok no issues. I will look for existing active forks before pushing next time. You can make our life easier by referencing issue number in PR message :)